### ISSUE TITLE
Tweepy 3.8.0 release timeline?

### ISSUE THREAD

**Issue by okomarov** [tweepy/tweepy#1232]
It's been a while since we had a bump in the release and a few PR have been merged in, with updated params, docs and fixes. 

Our requirements.txt points to a specific commit after the 3.7.0 release (with `git+https:...`) but a `pip freeze > requirements.txt` keeps resetting it to 3.7.0 when we update other requirements.

**Comment by Harmon758** [comment]
I'd like to get #1228 into v3.8.0. Currently, tests are failing for it because cassettes need to be updated.
I've contacted @joshthecoder about it, and I'm waiting on a response from him on what to do with them.
The PR could probably also use some testing.

For the documentation specifically, they're built after each commit and you can always access the latest version at https://tweepy.readthedocs.io/en/latest/.

As for your requirements.txt, a workaround might be installing it in editable mode, e.g. `-e git+https://github.com/tweepy/tweepy@3e0088a#egg=tweepy`, and using `pip freeze --all`.

**Comment by okomarov** [comment]
Thanks for the editable mode suggestion! Wasn't aware of it.

With respect to the features, #1228 seems pretty big and a feature that might require a few fixes/PRs before it reaches some stability (we don't use DM so it's an educated guess). 

The process of updating the cassettes is trivial but adjusting the tests will be more problematic since twitter has changed/deprecated some methods.

**Comment by Harmon758** [comment]
No problem.

I believe I covered all the aspects of the migration in the PR already, and it should be fully stable and reflective of and up to date with all the new methods.
I've also updated all the tests already and tested them locally with my own test account.

Currently, the issue with cassettes seems to be figuring out which account to use (https://github.com/tweepy/tweepy/commit/c880c907d4c302ff52ea6729f4307bec1c8069bb seems to have switched from @tweepytest to @TheTweepyTester while missing a few cassettes and without having updated the Travis CI credentials or default testing configuration) and retrieving the credentials for that account, but I agree that it shouldn't be difficult to update them once all that is sorted out.

Once I have the process of creating a release and updating everything related to doing so down (hopefully with v3.8.0, as I don't anticipate there being much to it that I'm not already aware of), I wouldn't be opposed to bugfix releases, i.e. v3.8.1, etc., if there are any immediate issues with any of the changes. Of course, I'd rather the PR be tested thoroughly beforehand, but I wouldn't wait half a year or more to release a simple bugfix with the next minor version.
